util/httputil/http_util.go

```diff
@@ -76,9 +76,11 @@ func newAuthRoundTripper(auth *config.Auth, rt http.RoundTripper) http.RoundTrip
 func CreateTransport(auth *config.Auth, transportConfig *http.Transport, timeout time.Duration) (http.RoundTripper, error) {
 	// Limits the time spent establishing a TCP connection if a new one is
 	// needed.
-	transportConfig.Dial = (&net.Dialer{
-		Timeout: timeout,
-	}).Dial
+	if transportConfig.Dial == nil && transportConfig.DialContext == nil {
```

Comment:
I'm seeing in my IDE that Dial is deprecated in favour of DialContext, so we can probably just remove Dial here and below, using just DialContext (explained here as well, line 140: https://golang.org/src/net/http/transport.go?s=13608:13676 )

Apart from that, I've tested and it's working fine for me

Reply:
@jotak should be ok now. I've changed so that we only create a new DialContext if both Dial and DialContext is not set (if it is set somewhere, we shouldn't change it).